TUM Campus App/MVGManager.swift

```diff
@@ -23,22 +25,39 @@ final class MVGManager: Manager, SimpleSingleManager {
         self.config = config
     }
     
-    func fetchSingle() -> Response<DataElement?> {
-        return fetch().flatMap { stations in
-            guard let station = stations.first else { return .successful(with: nil) }
-            return self.fetch(for: station).map { departures in
-                return DetailedStation(station: station, departures: departures)
-            }
+//    func fetch() -> Response<[Station]> {
+//        return config.mvg.doObjectsRequest(to: .getNearbyStations,
+//                                           queries: ["latitude" : location.coordinate.latitude,
+//                                                     "longitude" : location.coordinate.longitude],
+//                                           at: ["locations"]).map { $0.sorted(byLocation: \.location) }
+//    }
+    
+    func fetch() -> Response<[Station]> {
+        let promise: Response<[Station]> = config.mvg.doObjectsRequest(to: .getNearbyStations,
+                                                   queries: ["latitude" : location.coordinate.latitude,
+                                                             "longitude" : location.coordinate.longitude],
+                                                   at: ["locations"]).map { $0.sorted(byLocation: \.location) }
+        return promise.flatMap { (stations: [Station]) in
+            return self.fetchDepartures(for: stations)
         }
     }
     
-    func fetch() -> Response<[Station]> {
-        return config.mvg.doObjectsRequest(to: .getNearbyStations,
-                                           queries: ["latitude" : location.coordinate.latitude,
-                                                     "longitude" : location.coordinate.longitude],
-                                           at: ["locations"]).map { $0.sorted(byLocation: \.location) }
+    func fetchDepartures(for stations: [Station]) -> Response<[Station]> {
+        
+        let map = stations.flatMap {self.fetchDeparture(for: $0)}
+        
+        let bulk = map.bulk.onSuccess { return $0 }
+        return bulk
     }
     
+    func fetchDeparture(for data: Station) -> Response<Station> {
+        
+        return config.mvg.doJSONRequest(to: .departure, arguments: ["id" : data.id], queries: [ "footway": 0 ]).map { (json: JSON) in
+            let departures = json["departures"].array ==> Departure.init <** data
+            data.departures = departures.array(withFirst: 5)
+                return data
+        }
+    }
```

Comment:
@mathiasquintero I am sure this can be improved 😄

Reply:
@TG908 I believe I have improved it. It should even be discarding departure times that happened more than a minute ago

Reply:
Please check out the DetailedStation for more info.
This class is now in charge of making sure we're up to date concerning the current departure times